mumd/src/command.rs

```diff
@@ -47,13 +48,18 @@ pub async fn handle(
                 response_sender.send(generator()).unwrap();
             }
             ExecutionContext::Ping(generator, converter) => {
-                match generator() {
+                let ret = generator();
+                debug!("Ping generated: {:?}", ret);
+                match ret {
                     Ok(addr) => {
+                        let id = ping_count.fetch_add(1, Ordering::Relaxed);
                         let res = ping_request_sender.send((
-                            0,
+                            id,
                             addr,
                             Box::new(move |packet| {
-                                response_sender.send(converter(packet)).unwrap();
+                                if let Some(packet) = packet {
+                                    response_sender.send(converter(packet)).unwrap();
+                                }
```

Comment:
We probably want to always send something on the response channel, even if the request timed out, since I think it is worth maintaining the inner invariant that a processed command will always send a response sometime in the future.